BAZEL-bash.md

```diff
@@ -0,0 +1,162 @@
+# Working with Bash in Bazel
+
+Bazel has two out-of-the-box rules, `sh_test` and `genrule`, that can be used
+for one-shot custom build targets. If there is some reusable logic behind your
+target, consider building a new rule in Skylark instead.
+
+## Platform
+
+There are a number of limitations in the way Bazel will set uo the environment
+for these Bash scripts when running them, and they differ based on the OS. The
+Windows setup is the most constraining and the macOS one is the most
+permissive, i.e. when writing a new rule, the rule working on macOS does not
+give any guarantee that it will also run on either Linux or Windows, and the
+rule working on Linux does not guaranteee it will run on Windows. In my
+experience, it does seem like rules working on Windows will also work on Linux
+and macOS.
+
+Some of the default paths Bazel creates for you seem to depend on the platform,
+so be sure to always access dependencies through Bazel's `location` mechanism
+rather than rely on th specific directory structure it has created for you on
+your machine.
+
+## Dependencies
+
+Bazel promises hermeticity, but Bash scripts have access to whatever they want.
+As far as I can tell they are not run in any kind of chroot or container (at
+least on macOS), although they are run with a restricted PATH (on Linux) and in
+a specially set-up folder.
+
+As such, in order to get all of your dependencies available, you need to
+specify them in the relevant attributes: `srcs` for files required during
+build, `data` for files required at run time, and `tools` for executables. If
```

Comment:
`genrule` doesn’t have `data` at all. For `sh_test` the Bazel documentation states
```
This attribute (`srcs`) must be a singleton list, whose element is the shell script. This script must be executable, and may be a source file or a generated file. All other files required at runtime (whether scripts or data) belong in the data attribute. 
```
which should hopefully make clear what should go in `data` and in `srcs`.